presto-docs/src/main/sphinx/connector/kafka.rst

```diff
@@ -259,6 +264,430 @@ Field           Required  Type      Description
 
 There is no limit on field descriptions for either key or message.
 
+Kafka inserts
+-------------
+
+The Kafka connector supports the use of :doc:`/sql/insert` statements to write
+data to a Kafka topic. Table column data is mapped to Kafka messages as defined
+in the :ref:`table definition file<kafka-table-definition-files>`. There are
+four supported data formats for key and message encoding:
+
+* :ref:`raw format<raw-encoder>`
+* :ref:`CSV format<csv-encoder>`
+* :ref:`JSON format<json-encoder>`
+* :ref:`Avro format<avro-encoder>`
+
+These data formats each have an encoder that handles mapping table column values.
+Table column values are mapped by the encoders and sent to Kafka topics as bytes.
+
+Presto supports at-least-once delivery for Kafka producers. This means that
+messages are guaranteed to be sent to Kafka topics at least once. If a producer
+acknowledgement times out or if the producer receives an error it might retry
+sending the message. This could result in a duplicate message being sent to the
+Kafka topic.
+
+The Kafka connector does not allow the user to define which partition to send a
+message to. If a key is inserted with the message the producer will use a hash
+algorithm to decide which partition to send the message to. The same key will
+always be assigned the same partition.
+
+Row encoding
+------------
+
+Encoding allows :doc:`/sql/insert` statements to be used and defines how table
+columns in Presto map to Kafka key and message data.
+
+The Kafka connector contains the following encoders:
+
+* :ref:`raw encoder<raw-encoder>` - Table columns are mapped to Kafka
+  message as raw bytes
+* :ref:`CSV encoder<csv-encoder>` - Kafka message is formatted as comma
+  separated value
+* :ref:`JSON encoder<json-encoder>` - Table columns are mapped to JSON
+  fields
+* :ref:`Avro encoder<avro-encoder>` - Kafka message is mapped to Avro
+  fields based on an Avro schema
+
+.. note::
+
+    A :ref:`table definition file<kafka-table-definition-files>` must be defined
+    for the encoder to work.
+
+.. _raw-encoder:
```

Comment:
Use inline links for all of these. Care needs to be taken when adding references, as these names are global. In this case, we don't need them at all.